collect_app/src/main/java/org/odk/collect/android/provider/ProjectsProvider.kt

```diff
@@ -0,0 +1,48 @@
+package org.odk.collect.android.provider
+
+import android.content.ContentProvider
+import android.content.ContentValues
+import android.database.Cursor
+import android.database.MatrixCursor
+import android.net.Uri
+import org.odk.collect.android.injection.DaggerUtils
+import org.odk.collect.projects.ProjectsRepository
+import javax.inject.Inject
+
+class ProjectsProvider : ContentProvider() {
+
+    @Inject
+    lateinit var projectsRepository: ProjectsRepository
+
+    // Do not call it in onCreate() https://stackoverflow.com/questions/23521083/inject-database-in-a-contentprovider-with-dagger
+    private fun deferDaggerInit() {
+        DaggerUtils.getComponent(context).inject(this)
+    }
+
+    override fun onCreate(): Boolean {
+        return true
+    }
+
+    override fun query(uri: Uri, projection: Array<out String>?, selection: String?, selectionArgs: Array<out String>?, sortOrder: String?): Cursor? {
```

Comment:
It feels weird to me to ignore all of the parameters here. Is there another story to use them? If not, should there be some kind of error/warning if they're included?

Eventually the other project details will be included, right? I'm specifically thinking it will be important to include the server URI because that's a stable value that can be matched on. Imagine you want to write a companion app that reads form and submission metadata from a specific project only. You're going to need some way of consistently identifying the project you need. Doing so by server URI (and maybe also username) seems like the most reliable option (recognizing that there could be more than one project with the same server URI).

I also wonder whether we should support the id convention so that folks can easily display a list of projects in their companion apps (https://developer.android.com/guide/topics/providers/content-provider-creating). Or is your idea that you'd like people to explicitly request this before we build it? (fine by me)

Reply:
I think adding warnings and supporting other parameters should be an extra issue. I've added a discussion item to our catch-up tomorrow as then we walk through some different workflows that we might want to support.

 I'm definitely on board with a lot of this: I think adding projects to Collect's content provider could potentially allow for better integration with Collect (setting up projects for "workflow" apps) which might be a nice way to avoid people needing to fork the app.

Reply:
Yes, this should be a separate issue. We can discuss it once we are sure what project details should be included.